src/gazebo_mavlink_interface.cpp

```diff
@@ -392,7 +392,17 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 
     presetManager->GetCurrentProfileParam("max_step_size", param);
     const double max_step_size = our_any_cast<double>(param);
-    if (1.0 / real_time_update_rate != max_step_size)
+
+    // Adapt the real_time_update_rate according to the speed
+    // that we ask for in the env variable.
+    // Makes sure we don't do this multiple times.
+    if (1.0 / (real_time_update_rate / speed_factor_) != max_step_size) {
```

Comment:
I believe we are using the sanity check we were doing before as a way to prevent the same conditions being applied multiple times. However, could we not use the same conditions so that we can keep the sanity check as is?

Reply:
I bet that we could use the original sanity check if we applied the `speed_factor_` in a different way, rather than using my hacky sanity check, like you mentioned in the parent review.

Reply:
Resolving this specific conversation and tracking with the other one.